### PR TITLE
[MM_20423] 'external_image'module to ts (#16151)

### PR DESCRIPTION
**Pull request by dizkek**
#### Summary

Converted 'external_image' module to ts

#### Ticket Link
Fixes https://github.com/mattermost/mattermost-server/issues/16151

### CODE REVIEW

File: components/external_image/external_image.tsx
Comment (jespino) [inline components/external_image/external_image.tsx:9]: I think this `ImageMetaData` it is actually a `PostImage` object type from `mattermost-redux/types/posts` (that doesn't have the `format` field defined. So I think we should add that field into `mattermost-redux` repo.

In any case, I think the fields are not optional. The image metadata can exists or not, but if it exists, it should have all the fields. And actually, there is another field that is not shown here that is the `frame_count` for gif images.

Comment (dizkek) [reply components/external_image/external_image.tsx:9]: I checked `mattermost-redux/types/posts` and I reckon I can add format property in `PostImage ` and can import from there. Thanks for that information.

I reckon I can change width and height fields are not optional since type error only happens in `external_image_test` file when I change those properties are not optional. Hence, I can change teseCase.

But I am so confused whether I have to add format field as optional or not because I checked `post_attatchment_opengraph`(which uses`PostImage` type) but I cannot find any code has format filed for this type.

Comment (hahmadia) [reply components/external_image/external_image.tsx:9]: It's not optional. I checked our backend code and we have this structure defined there as well `PostImage` and you can see it has the following fields defined
image
This shows that the we need to add a `format: string` and `frameCount: number` to the PostImage type in redux.

The redux object should be mapping to the object defined in the backend (or in go terms, the struct)

File: components/external_image/external_image.tsx
Comment (hahmadia) [inline components/external_image/external_image.tsx:18]: Import `PostImage` from redux and use it here.
You can create a new PR which adds those above fields to the `PostImage` object in the redux repository.

**Review by hahmadia — COMMENTED** [review]
Looks good! Just some changes that need to be made in addition to comments Jesus made. Thanks!

File: components/external_image/external_image.tsx
Comment (dizkek) [reply components/external_image/external_image.tsx]: I changed my code and pushed again. Actually, I changed types as you advised me in the last comment. I checked type-test locally and it worked okay. But the problem is I changed types in mattermost-redux which is belongs to node_modules. Hence, its changes has not been updated in this commit and type-checks errors are keep failing.

Comment (dizkek) [reply components/external_image/external_image.tsx]: I requested PR in redux-repository and waiting for the merge.

Comment (hahmadia) [reply components/external_image/external_image.tsx]: The build failure is because the redux hash needs to be update with latest commit in your redux PR. 
Once your redux PR gets merged, then we can update the redux hash with the latest in master redux before merging this PR into master.

**Comment by hahmadia** [comment]
The type check failure you are seeing is not anything to do with your PR. The latest redux changes have not had their web app component updated so it's failing. I'll submit a separate PR for this and once it's merged into master, we can merge it into your PR and we should be good! Sorry about the inconvenience.

**Comment by hahmadia** [comment]
Awesome! Everything is good now. Approving :)

**Review by hahmadia — APPROVED** [review]

File: package.json
Comment (jespino) [inline package.json:40]: You can update this now to the master branch head in the mattermost-redux repo. Other than that, looks good to me. Thanks @dizkek!

**Review by jespino — CHANGES_REQUESTED** [review]
I tiny change required (related to the process), once is done, we can merge it.

**Comment by jespino** [comment]
The QA can be done now because the changes needed shouldn't affect the behavior.

**Review by saturninoabril — APPROVED** [review]
Thanks @dizkek for your contribution! LGTM, tested and passed.

File: package.json
Comment (dizkek) [reply package.json:40]: I changed but there is still some type errors....

**Comment by hahmadia** [comment]
To help push this over the finish line, I helped sort out the merge problems. The redux did not need to be updated as it's already been updated in the latest master.

**Comment by hahmadia** [comment]
@jespino Please review when you have time. Thanks

**Review by jespino — APPROVED** [review]
Looks good to me. Thanks @tbalthazar! :tada:

**Comment by dizkek** [comment]
Thanks so much guys for my first PR!

**Pull request merged**
